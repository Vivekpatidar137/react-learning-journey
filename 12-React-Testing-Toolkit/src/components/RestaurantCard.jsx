import { IMG_CDN_URL } from "./config";
import label from "/src/myAssets/veg.png";

const RestaurantCard = ({
  name,
  cloudinaryImageId,
  avgRatingString,
  cuisines,
  costForTwo,
}) => {
  return (
    <div
      data-testid="resCard"
      className="w-72 border border-gray-300 rounded-lg m-4 bg-white shadow-lg transition-transform transform hover:translate-y-[-0.125rem] relative"
    >
      <img
        src={IMG_CDN_URL + cloudinaryImageId}
        className="w-full h-52 object-cover rounded-t-lg"
      />
      <h2 className="text-lg font-semibold px-4 py-2">{name}</h2>
      <h3 className="text-sm text-gray-600 px-4">
        {avgRatingString} star â€¢ {costForTwo}
      </h3>
      <h4 className="text-sm text-gray-500 px-4 pb-4">
        {cuisines?.join(", ")}
      </h4>
    </div>
  );
};

export const withVegLabel = (RestaurantCard) => {
  return (props) => {
    return (
      <div className="relative">
        <label className="absolute top-0 left-4 z-10">
          <img src={label} alt="veg label" className="size-7 bg-slate-50" />
        </label>
        <RestaurantCard {...props} />
      </div>
    );
  };
};

export default RestaurantCard;
