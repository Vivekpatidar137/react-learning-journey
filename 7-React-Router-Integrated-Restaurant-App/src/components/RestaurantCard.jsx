import { IMG_CDN_URL } from "./config";

const RestaurantCard = ({
  name,
  cloudinaryImageId,
  avgRatingString,
  cuisines,
  costForTwo,
}) => {
  return (
    <div className="card">
      <img src={IMG_CDN_URL + cloudinaryImageId} />
      <h2>{name}</h2>
      <h3>
        {avgRatingString} star â€¢ {costForTwo}
      </h3>
      <h4>{cuisines?.join(", ")}</h4>
    </div>
  );
};

export default RestaurantCard;