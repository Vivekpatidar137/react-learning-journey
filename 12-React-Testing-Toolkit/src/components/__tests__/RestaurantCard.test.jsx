import { render, screen } from "@testing-library/react";
import RestaurantCard from "../RestaurantCard";
import MOCK_DATA from "../__mocks__/resCardMock.json";
import "@testing-library/jest-dom";
// Mock the IMG_CDN_URL value
jest.mock("../config", () => ({
  IMG_CDN_URL:
    "https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/",
}));

test("Should render RestaurantCard component", () => {
  render(<RestaurantCard {...MOCK_DATA.info} />);
  const name = screen.getByText("Ratlami Teekha");
  expect(name).toBeInTheDocument();
});

test("Should render the correct restaurant image", () => {
  render(<RestaurantCard {...MOCK_DATA.info} />);
  const image = screen.getByRole("img");

  // Compare with the mocked IMG_CDN_URL, not from MOCK_DATA
  expect(image).toHaveAttribute(
    "src",
    `https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/${MOCK_DATA.info.cloudinaryImageId}`
  );
});

test("Should display average rating and cost for two", () => {
  render(<RestaurantCard {...MOCK_DATA.info} />);
  const ratingCostText = screen.getByText(`${MOCK_DATA.info.avgRatingString} star â€¢ ${MOCK_DATA.info.costForTwo}`);
  expect(ratingCostText).toBeInTheDocument();
});
