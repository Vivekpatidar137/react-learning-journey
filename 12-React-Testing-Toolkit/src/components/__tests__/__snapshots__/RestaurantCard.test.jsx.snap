// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    class="w-72 border border-gray-300 rounded-lg m-4 bg-white shadow-lg transition-transform transform hover:translate-y-[-0.125rem] relative"
    data-testid="resCard"
  >
    <img
      class="w-full h-52 object-cover rounded-t-lg"
      src="https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/6af9332845d39b846914eb581f94fa76"
    />
    <div
      class="p-4"
    >
      <h2
        class="text-lg font-semibold"
      >
        Ratlami Teekha
      </h2>
      <h3
        class="text-sm text-gray-600"
      >
        4.0
         star • 
        ₹199 for two
      </h3>
      <div
        class="text-sm text-gray-500 overflow-hidden h-10"
      >
         
        Snacks
      </div>
    </div>
  </div>
</div>
`;
